halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/kubernetes/KubernetesAccountValidator.java

```diff
@@ -82,6 +85,46 @@ public void validate(ConfigProblemSetBuilder psBuilder, KubernetesAccount accoun
     }
   }
 
+  private void validateKindConfig(ConfigProblemSetBuilder psBuilder, KubernetesAccount account) {
+    List<String> kinds = account.getKinds();
+    List<String> omitKinds = account.getOmitKinds();
+    List<KubernetesAccount.CustomKubernetesResource> customResources = account.getCustomResources();
+
+    if (account.getProviderVersion() == Provider.ProviderVersion.V1) {
+      if (CollectionUtils.isNotEmpty(kinds) || CollectionUtils.isNotEmpty(omitKinds) || CollectionUtils.isNotEmpty(customResources)) {
+        psBuilder.addProblem(WARNING, "Kubernetes accounts at V1 do no support configuring caching behavior for kinds or custom resources");
+      }
+
+      return;
+    }
+
+    if (CollectionUtils.isNotEmpty(kinds) && CollectionUtils.isNotEmpty(omitKinds)) {
+      psBuilder.addProblem(ERROR, "At most one of \"kinds\" and \"omitKinds\" may be specified.");
+    }
+
+    if (CollectionUtils.isNotEmpty(kinds) && CollectionUtils.isNotEmpty(customResources)) {
+      List<String> unmatchedKinds = customResources.stream()
+          .map(KubernetesAccount.CustomKubernetesResource::getKubernetesKind)
+          .filter(cr -> !kinds.contains(cr))
+          .collect(Collectors.toList());
+
+      if (CollectionUtils.isNotEmpty(unmatchedKinds)) {
+        psBuilder.addProblem(WARNING, "The following custom resources \"" + customResources + "\" will not be cached since they aren't listed in your kinds \"" + kinds + "\".");
```

Comment:
"will not be cached since they aren't listed in your Kubernetes resource kinds configuration, which is ..."